ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -0,0 +1,596 @@
+ */
+package com.github.ambry.router;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.commons.ByteBufferAsyncWritableChannel;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.commons.ResponseHandler;
+import com.github.ambry.commons.ServerErrorCode;
+import com.github.ambry.config.RouterConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.network.NetworkClient;
+import com.github.ambry.network.NetworkClientErrorCode;
+import com.github.ambry.network.RequestInfo;
+import com.github.ambry.network.ResponseInfo;
+import com.github.ambry.protocol.RequestOrResponse;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+/**
+ * Tests for {@link GetBlobOperation}
+ */
+public class GetBlobOperationTest {
+  private static final int MAX_PORTS_PLAIN_TEXT = 3;
+  private static final int MAX_PORTS_SSL = 3;
+  private static final int CHECKOUT_TIMEOUT_MS = 1000;
+  private static final int BLOB_USER_METADATA_SIZE = 10;
+
+  private final MockTime time = new MockTime();
+  private final Map<Integer, GetOperation> correlationIdToGetOperation = new HashMap<>();
+  private final Random random = new Random();
+  private final FutureResult<ReadableStreamChannel> operationFuture = new FutureResult<>();
+
+  private int requestParallelism = 2;
+  private int successTarget = 1;
+  private int maxChunkSize;
+  private int blobSize = 1000;
+  private RouterConfig routerConfig;
+  private MockClusterMap mockClusterMap;
+  private NonBlockingRouterMetrics routerMetrics;
+  private MockServerLayout mockServerLayout;
+  private int replicasCount;
+  private AtomicReference<MockSelectorState> mockSelectorState = new AtomicReference<>();
+  private ResponseHandler responseHandler;
+  private MockNetworkClientFactory networkClientFactory;
+  private NetworkClient networkClient;
+  private VerifiableProperties vprops;
+  private NonBlockingRouter router;
+  private String blobIdStr;
+  private BlobProperties blobProperties;
+  private byte[] userMetadata;
+  private byte[] putContent;
+
+  private final GetTestRequestRegistrationCallbackImpl requestRegistrationCallback =
```

Comment:
or is it because you need to clear `requestList`? If that is the case, could you make that variable non-private. I know it doesn't matter accessibility-wise but while reading it makes it easier to understand it is used externally.